internal/store/peering_connections.go

```diff
@@ -98,11 +102,124 @@ func (s *Store) DeletePeeringConnection(projectID, peerID string) error {
 }
 
 // CreatePeeringConnection encapsulates the logic to manage different cloud providers.
-func (s *Store) CreatePeeringConnection(projectID string, peer *atlasv2.BaseNetworkPeeringConnectionSettings) (interface{}, error) {
+func (s *Store) CreatePeeringConnection(projectID string, peer *atlasv2.BaseNetworkPeeringConnectionSettings) (*atlasv2.BaseNetworkPeeringConnectionSettings, error) {
 	switch s.service {
 	case config.CloudService, config.CloudGovService:
 		result, _, err := s.clientv2.NetworkPeeringApi.CreatePeeringConnection(s.ctx, projectID, peer).Execute()
-		return result.GetActualInstance(), err
```

Comment:
This would have caused issues in the future where parent fields would not have been included in results.